### PR TITLE
Processing speed improvements

### PR DESCRIPTION
**Pull request by evanlombardo**
Increased performance of processing code. Originally used a ton of memory by storing all of the information as a list and took a long time to process the data since it was iterated over twice.

Now, data is stored in 50,000 line chunks (customizable) and processed by 4 threads (customizable).

Originally took around 2 minutes and 18 seconds to process 110 MB of data, now only takes around 1 minute and 10 seconds.

### CODE REVIEW

File: ner_processing/__main__.py [hunk 1]
```diff
@@ -38,39 +39,91 @@ def getLineCount(filepaths: List[str]) -> int:
...
+                            When stored data reaches specified amount, use threads to decode data faster
+                            """
 
+                            with multiprocessing.Pool(processors) as p:
+                                out = p.map(thread, lines)
+                                lines = []
+                            for data in out:
+                                for sub_data in data:
+                                    str_time = sub_data.timestamp.toString("yyyy-MM-ddTHH:mm:ss.zzzZ")
+                                    writer.writerow([str_time, sub_data.id, DATA_IDS[sub_data.id], sub_data.value])
```
Comment (rydzynskip) [inline ner_processing/__main__.py:111]: Throw these lines in a seperate function (something that looks like `process_lines(lines, writer)`) to remove the duplication.

**Review by rydzynskip — CHANGES_REQUESTED** [review]
Looks awesome. I ran some tests against larger files (100MB with over 2 million lines) and it reduced the time from ~9 minutes to ~4 minutes, and reduced RAM usage from 2.3 GB to 200 MB. 

A couple small changes and it's good to go. Well done

File: ner_processing/thread.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from ner_processing.decode_files import LogFormat, processLine
```
[the remark this answers is not in the document]
Comment (evanlombardo) [reply ner_processing/thread.py:1]: Actually, there is a bug with the multiprocessing library, when it handles non-native types, it throws an error unless the function used to thread is in another file.

File: ner_processing/__main__.py (hunk 1 above)
Comment (evanlombardo) [reply ner_processing/__main__.py]: This was a great suggestion, it forced me to use lines.clear(), which sped up the program by 3-4% with my test file.

File: ner_processing/thread.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rydzynskip) [reply ner_processing/thread.py:1]: Hmm weird, this is fine then

**Review by rydzynskip — APPROVED** [review]

**Pull request merged**